### ISSUE TITLE
Auto fill HTTP Auth: When multiple logins - No indication of HTTP auth and cannot enter manually

### ISSUE THREAD

**Issue by ProCycleDev** [keepassxreboot/keepassxc-browser#120]
##Expected Behavior

When you visit a site with a HTTP Auth dialog but there is more than one possible entry in the database a notification appears telling you it is waiting on HTTP authentication. There is a button in the addon icon bubble to summon the manual entry box.

## Current Behavior

When you visit a site with a HTTP Auth dialog but there is more than one possible entry in the database then the page just sits there loading forever. There is a notification that tells you "Multiple credentials detected" but it does not say you encountered an HTTP Auth dialog. Additionally if none of the login choices in the database are for the HTTP Auth then there is no way to manually enter a login.

## Possible Solution

An notification should appear telling you it is waiting on HTTP authentication. When you choose an entry from the addon icon bubble there should be an option to show the HTTP Auth Dialog.

## Steps to Reproduce (for bugs)

1. Add two entries to your database with the same url that asks for HTTP authentication. Ensure neither entry has the correct credientials
2. Attempt to visit the website.
3. Notification appears telling you there are multiple credentials detected
4. You cannot login because there is no way to show the manual entry dialog. If you didn't expect HTTP authentication then you sit there waiting for the page to load and it never does.

## Debug info

KeePassXC - 2.3.1
keepassxc-browser - 1.0.1
Operating system: Windows 7
Browser: Firefox
Proxy used: NO

**Comment by varjolintu** [comment]
I made some improvements to this and added a dismiss button to the popup dialog, plus modified the notification text. Basicly the dismiss button tries to add empty credentials to HTTP auth. Basically you could do the same by just selecting wrong credentials and then the manual popup arrives, but maybe a separate button is more clear :)

**Comment by gerritgriebel** [comment]
JFYI: I had exactly the described faulty behavior: Login form of a test site is protected by http basic auth. So I have two Logins, one for Login form and another one for basic auth. I can not login, it hangs after selecting the right login in basic auth attempt and I get above mentioned notification. My workaround was to disable HTTP Auth in Chrome Extension and it store HTTP-Password in Chrome Passwords.  I'm using todays latest versions, as I switched today to KeePassXC:

KeePassXC - 2.5.4
KeePassXC-Browser - 1.6.2
Operating system: MacIntel
Browser: Chrome/Chromium 81.0.4044.122

**Comment by varjolintu** [comment]
@gerritgriebel Does it have any difference if you restrict the HTTP Auth credential entry just for that? It's possible when you edit the entry and go to Browser Integration page.

**Comment by gerritgriebel** [comment]
No, only change is that I got offered only the entry which I marked for HTTP Auth only and not the other one. Still page hangs and I get the "Multiple credentials detected" notification.

**Comment by varjolintu** [comment]
And what the popup does when you click it?

**Comment by gerritgriebel** [comment]
When navigating to basic auth protected site, I get:
<img width="537" alt="Bildschirmfoto 2020-04-27 um 16 27 38" src="https://user-images.githubusercontent.com/3919675/80383721-40b77380-88a4-11ea-8407-af879f636d29.png"> and click "Allow".  Then page hangs (keeps on loading) and notification appears "Multiple logins, click on the browsers plugin icon" although german translation is too long for the notification popup and can only be closed:
<img width="379" alt="Bildschirmfoto 2020-04-27 um 16 11 53" src="https://user-images.githubusercontent.com/3919675/80382991-4a8ca700-88a3-11ea-9b5f-9f09a935c1cb.png">
When clicking on the browsers plugin icon as instructed I can select the right login and then http basic auth is executed. So it now works, but it's a rather cumbersome procedure involving three dialogs. My workaround with Chrome http basic popup involves just one keypress on enter key.

**Comment by varjolintu** [comment]
That is a known issue yes. HTTP Basic Auth doesn't have a DOM content on the page so the only way to select correct credentials is to use the popup. However, there's already a [PR for another method](https://github.com/keepassxreboot/keepassxc-browser/pull/782) that creates an additional window where the HTTP Basic Auth credentials can be used. It's not finished yet though.

You can just ignore the notification, or disable it for HTTP Basic Auth from the settings.
